spec/requests/admin_timeout_spec.rb

```diff
@@ -7,38 +7,49 @@
 
   let!(:user) { create(:dfe_signin_user, dfe_sign_in_id: dfe_sign_in_id) }
 
+  let(:before_expiry) { timeout_length_in_minutes.minutes - 2.seconds }
+  let(:after_expiry) { timeout_length_in_minutes.minutes + 1.second }
+
   before do
     sign_in_to_admin_with_role(DfeSignIn::User::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE, dfe_sign_in_id, organisation_id)
   end
 
-  context "no actions performed for more than the timeout period" do
-    let(:after_expiry) { timeout_length_in_minutes.minutes + 1.second }
-
-    it "clears the session and redirects to the login page" do
-      expect(session[:user_id]).to eq(user.id)
+  it "clears the session and redirects to the login page when no actions have been performed during the timeout period" do
+    expect(session[:user_id]).to eq(user.id)
 
-      travel after_expiry do
-        get admin_claims_path
+    travel after_expiry do
+      get admin_claims_path
 
-        expect(response).to redirect_to(admin_sign_in_path)
-        expect(session[:user_id]).to be_nil
+      expect(response).to redirect_to(admin_sign_in_path)
+      expect(session[:user_id]).to be_nil
 
-        follow_redirect!
-        expect(response.body).to include("Your session has timed out due to inactivity")
-      end
+      follow_redirect!
+      expect(response.body).to include("Your session has timed out due to inactivity")
     end
   end
 
-  context "no action performed just within the timeout period" do
-    let(:before_expiry) { timeout_length_in_minutes.minutes - 2.seconds }
+  it "does not extend the session when the user performs a claim action" do
+    travel before_expiry do
```

Comment:
Do we need to travel to before expiry here? Would calling `start_student_loans_claim` inside the `travel after_expiry` block below would still provide a failing spec for the bug? If so that would make it easier to follow and also closer to the behaviour that triggered the bug in the first place.

Reply:
Yeah, that's clearer 👍